### ISSUE TITLE
DTZ011 flags `datetime.date.today()` unnecessarily for pure date usage

### ISSUE THREAD

**Issue by Jason-Francis-github** [astral-sh/ruff#21661]
#### **Description**
The `DTZ011` rule currently flags `datetime.date.today()` as a violation, suggesting the use of `datetime.datetime.now(tz=...).date()` instead. While this makes sense for timezone-aware **datetime** operations, it feels overly strict when the requirement is for a **date only**, without any time component.

For example:

```python
from datetime import date

registration_date = date.today().strftime("%d/%m/%Y")
```

This is a common and reasonable pattern when working with dates in isolation (e.g., formatting for display, storing a date-only field). Time zones are irrelevant in this context because there is no ambiguity about hours or minutes.

---

#### **Why this matters**
- Enforcing timezone awareness for pure dates adds unnecessary complexity.
- The suggested alternative (`datetime.now(tz=...).date()`) introduces overhead without practical benefit for date-only use cases.
- Many projects (especially tests or business logic dealing with dates) will need to add `# noqa: DTZ011` or disable the rule entirely, reducing the value of linting.

---

#### **Proposed Solutions**
1. **Allow `date.today()` by default**  
   Exempt `datetime.date.today()` from `DTZ011` since it does not suffer from the same ambiguity as naive `datetime`.

2. **Introduce a separate rule or configuration**  
   - Keep `DTZ011` for `datetime.now()` and similar datetime operations.
   - Add an option to ignore date-only functions like `date.today()`.

3. **Document rationale clearly**  
   If the rule is intended to apply universally, clarify why `date.today()` should be considered problematic.

---

#### **Example of current behavior**
```python
# Current flagged usage:
from datetime import date
registration_date = date.today().strftime("%d/%m/%Y")  # noqa: DTZ011

# Suggested alternative (adds unnecessary complexity for date-only):
from datetime import datetime
from zoneinfo import ZoneInfo
registration_date = datetime.now(ZoneInfo("Europe/London")).date().strftime("%d/%m/%Y")
```

---

#### **Expected behavior**
No lint error for `date.today()` when used for date-only formatting or logic.

**Comment by MichaReiser** [comment]
Hi. `DTZ011` only flags `date.today()` usages. I think you can just disable the rule if, for your use case, the time zone part isn't important.

> Introduce a separate rule or configuration

This is already the case today. `datetime.now` is [`DTZ005`](https://docs.astral.sh/ruff/rules/call-datetime-now-without-tzinfo/)

**Comment by gshannon** [comment]
This answer misses the point entirely.  "I think you can just disable the rule" -- adds no information, OP already stated this in the context of a legitimate concern -- having to disable linting reduces its value, and I would add it's also annoying and adds code clutter.  You don't address that concern.  The other rule DTZ005 is irrelevant.  OP's suggested separate rule would have to address this issue, which DTZ005 does not.

**Comment by Jason-Francis-github** [comment]
Please do not close this. The response misses the point.
